### PR TITLE
[minor] SCA

### PR DESCRIPTION
**Pull request by kalessil**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | n/a
| License       | MIT
| Doc PR        | n/a

- Control flow tweaks

### CODE REVIEW

**Review by ostrolucky — APPROVED** [review]

File: src/Symfony/Component/Validator/Mapping/Factory/LazyLoadingMetadataFactory.php [hunk 1]
```diff
@@ -162,10 +162,6 @@ public function hasMetadataFor($value)
 
         $class = ltrim(is_object($value) ? get_class($value) : $value, '\\');
 
-        if (class_exists($class) || interface_exists($class)) {
-            return true;
-        }
-
-        return false;
+        return class_exists($class) || interface_exists($class);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Validator/Mapping/Factory/LazyLoadingMetadataFactory.php]: while at it, the second call should have false as second argument

File: src/Symfony/Component/Console/Input/InputOption.php [hunk 2]
```diff
@@ -56,8 +56,7 @@ public function __construct($name, $shortcut = null, $mode = null, $description
             if (is_array($shortcut)) {
                 $shortcut = implode('|', $shortcut);
             }
-            $shortcuts = preg_split('{(\|)-?}', ltrim($shortcut, '-'));
-            $shortcuts = array_filter($shortcuts);
+            $shortcuts = array_filter(preg_split('{(\|)-?}', ltrim($shortcut, '-')));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Console/Input/InputOption.php]: Not sure about this.

Comment (kalessil) [reply src/Symfony/Component/Console/Input/InputOption.php]: No performance gain, just fewer lines of code to maintain.

Comment (webignition) [reply src/Symfony/Component/Console/Input/InputOption.php]: In general, performing two distinct operations in two lines of code is easier to understand than performing two distinct operations in one line of code.

The number of lines of code to be maintained is less of an issue than the ability for the code to be easily understood.

I'd say this specific change is better being reverted to the two line version.

**Review by nicolas-grekas — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @kalessil.

**Pull request merged**